demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,357 @@
+            }
+        }
+
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~create inf req~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        std::vector<InferenceEngine::InferRequest> inferRequests;
+        int infReqNum = FLAGS_nireq;
+        for(int infReqID = 0; infReqID < infReqNum; ++infReqID)
+            inferRequests.push_back(executableNetwork.CreateInferRequest());
+
+       
+
+        std::queue<cv::Mat> showMats;
+        
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::mutex showMutex;
+
+        //curImg is stored in ieWrapper
+        int batchSize = FLAGS_b;
+        int irFirstIndex = 0;
+        
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~set complition callback~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        for (auto it = inferRequests.begin(); it != inferRequests.end(); ++it) {
+            it->SetCompletionCallback(
+                infReqCallback(*it, mutex, condVar, captures, showMats, irFirstIndex)
+            );
+            irFirstIndex = (irFirstIndex + batchSize) % inputImgs.size();
+        }
+        
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~filling blobs~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        int imgDataSize = inputImgs.size();
+        auto blobDims = inputBlobsDimsInfo[inputBlobName];
+        if (blobDims.size() != 4) {
+            throw std::runtime_error("Input data does not match size of the blob");
+        }
+        for (auto it = inferRequests.begin(); it != inferRequests.end(); ++it) { 
+            //setInputBlob(*it, curPos);    
+            auto inputBlob = it->GetBlob(inputBlobName);
+
+            for(int i = 0; i < batchSize; i++) {        
+                cv::Mat inputImg = inputImgs.at((captures.curPos+i)%imgDataSize);
+                matU8ToBlob<uint8_t>(inputImg, inputBlob, i);
+            }
+            captures.curPos = (captures.curPos + batchSize) % imgDataSize;
+        }
+
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~create output info~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        int width;
+        int height;
+        if(FLAGS_res != "") {
+            std::vector<std::string> gmRowsCols = split(FLAGS_res, 'x');        
+            if(gmRowsCols.size() != 2) throw std::runtime_error("Input gmsizes key is not valid.");
+            else{
+                width = std::stoi(gmRowsCols[0]);
+                height = std::stoi(gmRowsCols[1]);
+            }
+        } else {
+            width = 1080;
+            height =  1920;
+        }
+        GridMat gridMat = GridMat(cv::Size(width, height));
+        cv::namedWindow("main");
+        cv::imshow("main", gridMat.getMat());
+
+        //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~start async~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        int64 startTime = cv::getTickCount();
+        for (auto it = inferRequests.begin(); it != inferRequests.end(); ++it)
```

Comment:
I think it will be easier to read a range based for loop:
`for (InferRequest& inferRequest : inferRequests)`